org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/editor/copyqualifiedname/XbaseCopyQualifiedNameService.java

```diff
@@ -0,0 +1,197 @@
+		return toQualifiedName(jvmExecutable);
+	}
+
+	protected String _getQualifiedName(JvmExecutable jvmExecutable, Void context) {
+		return toQualifiedName(jvmExecutable);
+	}
+
+	protected String _getQualifiedName(JvmExecutable executable, XAbstractFeatureCall featureCall) {
+		IResolvedTypes resolvedTypes = typeResolver.resolveTypes(featureCall);
+		IResolvedExecutable resolvedExecutable = resolveExecutable(executable, featureCall, resolvedTypes);
+		return toFullyQualifiedName(executable) //
+				+ "(" + toQualifiedNames(featureCall.getActualArguments(), //
+						(argument) -> toQualifiedName(argument, resolvedExecutable, executable, resolvedTypes,
+								featureCall.getActualArguments()))
+				+ ")";
+	}
+
+	protected String _getQualifiedName(JvmConstructor constructor, XConstructorCall constructorCall) {
+		IResolvedTypes resolvedTypes = typeResolver.resolveTypes(constructorCall);
+		IResolvedExecutable resolvedExecutable = resolveExecutable(constructor, constructorCall, resolvedTypes);
+		return toFullyQualifiedName(constructor) //
+				+ "(" + toQualifiedNames(constructorCall.getArguments(), //
+						(argument) -> toQualifiedName(argument, resolvedExecutable, constructor, resolvedTypes,
+								constructorCall.getArguments()))
+				+ ")";
+	}
+
+	protected IResolvedExecutable resolveExecutable(JvmExecutable consturctor, XExpression expression, IResolvedTypes resolvedTypes) {
+		if (consturctor instanceof JvmConstructor && expression instanceof XAbstractFeatureCall) {
+			return _resolveExecutable((JvmConstructor) consturctor, (XAbstractFeatureCall) expression, resolvedTypes);
+		} else if (consturctor instanceof JvmConstructor && expression instanceof XConstructorCall) {
+			return _resolveExecutable((JvmConstructor) consturctor, (XConstructorCall) expression, resolvedTypes);
+		} else if (consturctor instanceof JvmOperation && expression instanceof XAbstractFeatureCall) {
+			return _resolveExecutable((JvmOperation) consturctor, (XAbstractFeatureCall) expression, resolvedTypes);
+		} else {
+			throw new IllegalArgumentException(
+					"Unhandled parameter types: " + Arrays.asList(consturctor, expression, resolvedTypes).toString());
+		}
+	}
+
+	protected IResolvedExecutable _resolveExecutable(JvmConstructor consturctor, XAbstractFeatureCall featureCall,
+			IResolvedTypes resolvedTypes) {
+		XExpression actualReceiver = featureCall.getActualReceiver();
+		LightweightTypeReference actualType = null;
+		if (actualReceiver != null) {
+			actualType = resolvedTypes.getActualType(actualReceiver);
+		}
+
+		ResolvedFeatures resolvedFeatures = null;
+		if (actualType != null) {
+			resolvedFeatures = overrideHelper.getResolvedFeatures(actualType);
+		}
+
+		List<IResolvedConstructor> declaredConstructors = null;
+		if (resolvedFeatures != null) {
+			declaredConstructors = resolvedFeatures.getDeclaredConstructors();
+		}
+
+		Optional<IResolvedConstructor> resolvedConstructor = FluentIterable.from(declaredConstructors)
```

Comment:
`FluentIterable` - A bit of inspiration from @szarnekow code.

Reply:
Does this work if declaredConstructors is null?

Reply:
`declaredConstructors` can be empty, but not null. PS - https://bit.ly/3dIG53z

Generally speaking, `FluentIterable.from` will throw a NPE if the passed param. is null.

Reply:
declaredConstructors will still be empty if resolvedFeatures is null.

These null checks look a little too xtendy to me. Many of the intermediate local vars will never be null if the actual receiver was present.
Please reverse the logic and exit early but otherwise keep a normal flow.

Reply:
Also my preference would always be something like
```
for(IResolvedConstructor candidate: declaredConstructors) {
  if (constructor.equals(candidate.getDeclaration()) {
    return candidate;
 }
}
return null

```

Reply:
It is going to be null if `resolvedFeatures` is null

Reply:
You are right, I missed that.